libs/SmartStore/SmartStoreTests/SFSmartSqlTests.m

```diff
@@ -245,6 +245,11 @@ - (void) testConvertSmartSqlWithQuotedUnbalancedCurlyBrace {
                           [self.store convertSmartSql:@"select {employees:education} from {employees} where {employees:education} like ' { { { } } '"]);
 }
 
+- (void) testConvertOtherComplexSmartSql {
```

Comment:
I added the same test on Android (see https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2234/commits/66d95aecacf263e21977ccb46fdd600e47c34c58).
It passes there without code changes.